cpp/src/rolling/rolling.cu

```diff
@@ -131,8 +135,16 @@ void gpu_rolling(gdf_size_type nrows,
     gdf_index_type const out_mask_location = cudf::util::detail::bit_container_index<bit_mask_t, gdf_index_type>(i);
 
     // only one thread writes the mask
-    if (0 == threadIdx.x % warpSize)
+    if (0 == threadIdx.x % warpSize) {
       out_col_valid[out_mask_location] = result_mask;
+      // Perform null bitmask null count
+      valid_count_accumulate += cudf::detail::single_lane_popc_block_reduce(result_mask);
+    }
+
+    //one atomicAdd per block
+    if (blockIdx.x == 0) {
```

Comment:
Likewise, this doesn't work either -- you are having every thread of block zero perform the atomic, rather than the first warp of every block. Should be `if (threadIdx.x == 0)`.